### ISSUE TITLE
How to use same side menu for all viewcontrollers and to viewdidload only once for each viewcontroller

### ISSUE THREAD

**Issue by alientyc** [jonkykong/SideMenu#183]
Hi,

I have read the [guidelines for contributing](https://github.com/jonkykong/SideMenu/blob/master/.github/CONTRIBUTING.md) and I understand:
- [x ] My issue is happening in the **latest version** of SideMenu.
- [ x] My issue was **not** solved in the [README](https://github.com/jonkykong/SideMenu/blob/master/README.md).
- [ x] My issue can **not** be answered on [stackoverflow.com](stackoverflow.com).
- [x ] My issue is **not** a request for new functionality that I am unwilling to build and contribute with a pull request.
- [x](url) My issue **is** reproducible in the [demo project] (https://github.com/jonkykong/SideMenu/tree/master/Example).

## Issue Description

I am trying to use the same side menu for mainviewcontroller and create a 2nd VC on the demo project. See screenshot attached
![screen shot 2017-04-24 at 5 07 44 pm](https://cloud.githubusercontent.com/assets/4916438/25330237/8b9325b6-2911-11e7-877a-5df8502484d0.png)

With this setup, i am able to click on the side menu to either display the demo project mainviewcontroller and the 2nd VC. But whenever i choose to display the 2nd VC, viewdidload executes which mean the view is 'unloaded'. How can I set it to make the 2nd VC viewdidload once only? I just need to use the viewdidappear to do other data display etc for 2nd VC.

I tried 'menuPushStyle: MenuPushStyle = .defaultBehavior, - popWhenPossible, preserve, preserveAndHideBackButton' but all still cause the 2nd VC to viewdidload everytime i select it.

Appreciate your advice.
Thanks.

**Comment by jorgecsan** [comment]
As I understand you want to open a new VC from the menu. If you create the 2nd VC with a Navigation Controller (as the menu is created) you can display a new ViewController (over the first one).

Hope it helps!!:)

**Comment by jonkykong** [comment]
This question is better suited for stackoverflow.com.

**Comment by alientyc** [comment]
Hi Jonkykong and jorgecsan,

Sorry if this not the right place to ask as I was wondering if it is an issue when i encounter this.

I am able to set up the 2 cells on the menu which will open the 2 VCs base on what i tap but trying to find out if i can keep the 'state of data' on the VC.

To know how many times the VCs are loaded/appear into the view, i inserted 'print to console' into viewdidload and viewdidappear of the 2 VCs.
I set VC 1 as the root view controller. If i tab VC 1 on the menu,  it will only 'print viewdidload+viewdidappear' once and just 'print viewdidappear the next time I tab it. This is correct and what i need similar like a 'tab controller feature'. However, for VC 2, it will print 'viewdidload+viewdidappear' all the time which means the view keeps 'reloading'. But i would like to keep the 'state of data' just like VC1.

So if i switch VC 2 as the rootviewcontroller, then VC 2 will keep the 'state of data' and VC 1 will not. I tried 'menuPushStyle: MenuPushStyle = .defaultBehavior, - popWhenPossible, preserve, preserveAndHideBackButton' but can't make it work.
I created a question on stackoverflow http://stackoverflow.com/questions/43703679/using-sidemenu-cocoapods-and-to-keep-the-state-of-data-for-all-vcs

Appreciate your advice.
Thanks.
